scaladoc-testcases/src/tests/thisType.scala

```diff
@@ -0,0 +1,8 @@
+package tests
+package thisType
+
+// issue 16024
+
+class X[Map[_, _[_]]]:
+  inline def map[F[_]](f: [t] => t => F[t]): Map[this.type, F] = //expected: inline def map[F[_]](f: [t] => (x$1: t) => F[t]): Map[this.type, F]
```

Comment:
More nitpicking:
Would it be possible to only print the identifier (`x$1`) if it is required ?

Ex: `[t] => (x: t) => t`

I see two ways of doing so:
1. Look at whether the name is synthetic/mangled: ex prints as `[t] => (x: t) => t`
1. Look at whether the name is used on the rhs: ex prints as `[t] => t => t`

Reply:
I'll look into that.

Reply:
Let's leave that for future improvements.

Reply:
Could you open an issue for it ?

Reply:
Done: https://github.com/lampepfl/dotty/issues/17756